src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Processors/FilteringTelemetryProcessor.cs

```diff
@@ -49,7 +49,7 @@ private bool IsEnabled(ITelemetry item)
                 if (telemetry.Properties.TryGetValue(LogConstants.LogLevelKey, out string logLevelString) &&
                     Enum.TryParse(logLevelString, out LogLevel logLevel))
                 {
-                    LoggerFilterRule filterRule = _ruleMap.GetOrAdd(categoryName, SelectRule(categoryName));
```

Comment:
This was a big one - simple mistake meaning we were running the rule selection logic for every log